src/Microsoft.AspNet.Mvc.TagHelpers/CacheTagHelper.cs

```diff
@@ -0,0 +1,266 @@
+        /// <summary>
+        /// Gets or sets the name of a HTTP request header to vary the cached result by.
+        /// </summary>
+        [HtmlAttributeName(VaryByHeaderAttributeName)]
+        public string VaryByHeader { get; set; }
+
+        /// <summary>
+        /// Gets or sets a comma-delimited set of query parameters to vary the cached result by.
+        /// </summary>
+        [HtmlAttributeName(VaryByQueryAttributeName)]
+        public string VaryByQuery { get; set; }
+
+        /// <summary>
+        /// Gets or sets a comma-delimited set of route data parameters to vary the cached result by.
+        /// </summary>
+        [HtmlAttributeName(VaryByRouteAttributeName)]
+        public string VaryByRoute { get; set; }
+
+        /// <summary>
+        /// Gets or sets a comma-delimited set of cookie names to vary the cached result by.
+        /// </summary>
+        [HtmlAttributeName(VaryByRouteAttributeName)]
+        public string VaryByCookie { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value that determines if the content is to be varied by the Identity for the logged in 
+        /// <see cref="HttpContext.User"/>.
+        /// </summary>
+        [HtmlAttributeName(VaryByUserAttributeName)]
+        public bool VaryByUser { get; set; }
+
+        /// <summary>
+        /// Gets or sets the exact <see cref="DateTimeOffset"/> the cache entry should be evicted.
+        /// </summary>
+        [HtmlAttributeName(ExpiresOnAttributeName)]
+        public DateTimeOffset? ExpiresOn { get; set; }
+
+        /// <summary>
+        /// Gets or sets the duration from the time the cache entry when it should be evicted.
+        /// </summary>
+        [HtmlAttributeName(ExpiresAfterAttributeName)]
+        public TimeSpan? ExpiresAfter { get; set; }
+
+        /// <summary>
+        /// Gets or sets the duration from last access that the cache entry should be evicted.
+        /// </summary>
+        [HtmlAttributeName(ExpiresSlidingAttributeName)]
+        public TimeSpan? ExpiresSliding { get; set; }
+
+        /// <summary>
+        /// Gets or sets the <see cref="CachePreservationPriority"/> policy for the cache entry.
+        /// </summary>
+        [HtmlAttributeName(CachePriorityAttributeName)]
+        public CachePreservationPriority? Priority { get; set; }
+
+        /// <inheritdoc />
+        public override async Task ProcessAsync(TagHelperContext context, TagHelperOutput output)
+        {
+            var key = GenerateKey(context);
+            string result;
+            if (!MemoryCache.TryGetValue(key, out result))
+            {
+                result = await context.GetChildContentAsync();
+                MemoryCache.Set(key, cacheSetContext =>
+                {
+                    UpdateCacheContext(cacheSetContext);
+                    return result;
+                });
+            }
+
+            // Clear the contents of the "cache" element since we don't want to render it.
```

Comment:
The end goal is to not clear the contents, it's to clear the tag itself. Also, keep in mind by calling SupressOutput here you're essentially doing: 
```C#
TagName = null;
PreContent = null;
Content = null;
PostContent = null;
```

Meaning, any other `TagHelper` that targets the cache tag will have their `PreContent`/`PostContent` nuked, i'm not sure that's what we want to do. Also, I think it's worth mentioning that if the `output.Content` is already set at the beginning of the `ProcessAsync` (set by another `TagHelper`) method that we may want to consider no-oping; that's what we do for our other oob `TagHelper`s. /cc @DamianEdwards